app/code/Magento/Catalog/Model/Product/Option.php

```diff
@@ -135,14 +152,34 @@ public function __construct(
         \Magento\Framework\Model\ResourceModel\AbstractResource $resource = null,
         \Magento\Framework\Data\Collection\AbstractDb $resourceCollection = null,
         array $data = [],
-        ProductCustomOptionValuesInterfaceFactory $customOptionValuesFactory = null
+        ProductCustomOptionValuesInterfaceFactory $customOptionValuesFactory = null,
+        array $optionGroups = [],
+        array $optionGroupsToTypes = []
     ) {
         $this->productOptionValue = $productOptionValue;
         $this->optionTypeFactory = $optionFactory;
-        $this->validatorPool = $validatorPool;
         $this->string = $string;
+        $this->validatorPool = $validatorPool;
         $this->customOptionValuesFactory = $customOptionValuesFactory ?:
             \Magento\Framework\App\ObjectManager::getInstance()->get(ProductCustomOptionValuesInterfaceFactory::class);
+        $this->optionGroups = $optionGroups ?: [
+            self::OPTION_GROUP_DATE => Date::class,
+            self::OPTION_GROUP_FILE => File::class,
+            self::OPTION_GROUP_SELECT => Select::class,
+            self::OPTION_GROUP_TEXT => Text::class,
+        ];
+        $this->optionGroupsToTypes = $optionGroupsToTypes ?: [
```

Comment:
Once you added constants - it’s became clear that it’s actually opposite mapping, option type to group, please rename the property

Sorry that didn’t realized it earlier

Reply:
Also sorry for my inattention. You spent on my PR too much time.